### PR TITLE
Recipe

### PR DESCRIPTION
**Pull request by AlexandreMELIN**

### CODE REVIEW

File: Domain/src/test/java/miscellaneous/JavaFeaturesTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class JavaFeaturesTest {
+    @Test
+    void ensureThatForEachOperatorMaintainsTheRightOrderAfterASortOperation() {
```
Comment (Ninjeneer) [inline Domain/src/test/java/miscellaneous/JavaFeaturesTest.java:13]: Javascript veteran spotted right here

File: Domain/src/main/java/recipe/usecase/update/UpdateRecipeUseCase.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    private final RecipeUpdater updater;
+    private final RecipePresenceChecker presenceChecker;
+
+    public UpdateRecipeUseCase(RecipeUpdater updater, RecipePresenceChecker presenceChecker) {
+        this.updater = updater;
+        this.presenceChecker = presenceChecker;
+    }
+
+    public void execute(UpdateRecipeRequest request, UpdateRecipePresenter presenter) throws IllegalArgumentException {
+        if (!presenceChecker.aRecipeWithThisNameAlreadyExists(request.oldRecipeName())) {
```
Comment (Ninjeneer) [inline Domain/src/main/java/recipe/usecase/update/UpdateRecipeUseCase.java:19]: We should also test if we are not overriding an existing recipe name
Or if we take the name as the identifier key, then good to merge

Comment (AlexandreMELIN) [reply Domain/src/main/java/recipe/usecase/update/UpdateRecipeUseCase.java:19]: Depend on how we want to handle things. Can we have 2 recipes with the same name?

File: Domain/src/test/java/miscellaneous/JavaFeaturesTest.java (hunk 1 above)
Comment (AlexandreMELIN) [reply Domain/src/test/java/miscellaneous/JavaFeaturesTest.java:13]: We're never too sure

File: Domain/src/main/java/recipe/usecase/update/UpdateRecipeUseCase.java (hunk 2 above)
Comment (Ninjeneer) [reply Domain/src/main/java/recipe/usecase/update/UpdateRecipeUseCase.java:19]: I guess no. But I am afraid of parsing issues in the API urls

**Pull request merged**